packages/search/tests/unit/services/navi-search/navi-asset-search-provider-test.js

```diff
@@ -0,0 +1,110 @@
+    const store = this.owner.lookup('service:store'),
+      mockAuthor = store.createRecord('user', { id: 'ciela' });
+    this.owner.register(
+      'service:user',
+      Service.extend({
+        getUser: () => mockAuthor
+      })
+    );
+  });
+
+  test('construct full search query for reports', function(assert) {
+    assert.deepEqual(
+      service._constructSearchQuery('Hyrule', 'navi_user', 'report'),
+      {
+        filter: { reports: '(title==*Hyrule*,request==*Hyrule*);author==*navi_user*' }
+      },
+      'Constructs the correct report query for the api with both filter parameters and author.'
+    );
+  });
+
+  test('construct only query parameters search query for reports', function(assert) {
+    assert.deepEqual(
+      service._constructSearchQuery('Hyrule', null, 'report'),
+      {
+        filter: { reports: '(title==*Hyrule*,request==*Hyrule*)' }
+      },
+      'Constructs the correct report query for the api with filter parameters.'
+    );
+  });
+
+  test('construct only author search query for reports', function(assert) {
+    assert.deepEqual(
+      service._constructSearchQuery(null, 'navi_user', 'report'),
+      {
+        filter: { reports: 'author==*navi_user*' }
+      },
+      'Constructs the correct report query for the api with author.'
+    );
+  });
+
+  test('construct full search query for dashboards', function(assert) {
+    assert.deepEqual(
+      service._constructSearchQuery('Hyrule', 'navi_user', 'dashboard'),
+      {
+        filter: { dashboards: '(title==*Hyrule*);author==*navi_user*' }
+      },
+      'Constructs the correct dashboard query for the api with both filter parameters and author.'
+    );
+  });
+
+  test('construct only query parameters search query for dashboards', function(assert) {
+    assert.deepEqual(
+      service._constructSearchQuery('Hyrule', null, 'dashboard'),
+      {
+        filter: { dashboards: '(title==*Hyrule*)' }
+      },
+      'Constructs the correct report query for the api with filter parameters.'
+    );
+  });
+
+  test('construct only author search query for dashboards', function(assert) {
+    assert.deepEqual(
+      service._constructSearchQuery(null, 'navi_user', 'dashboard'),
+      {
+        filter: { dashboards: 'author==*navi_user*' }
+      },
+      'Constructs the correct dashboard query for the api with author.'
+    );
+  });
+
+  test('search by user search returns reports and dashboards', async function(assert) {
+    const results = await service.search.perform('Revenue');
+    assert.equal(results.component, 'navi-search-result/asset', 'Result contains correct display component name');
+    assert.equal(results.title, 'Reports & Dashboards', 'Result contains correct title for the search result section');
+    results.data.forEach(async function(result) {
+      let author = await result.author;
+      assert.ok(result.title.includes('Revenue'), 'The service returns a report that includes the requested title.');
```

Comment:
It seems like we support searching by title, request content, and author, but we only have mock results that are matched based on title and author. Can we add a test to make sure we're querying against request content correctly?